pkg/validate/openshiftcluster_validatedynamic.go

```diff
@@ -159,10 +164,22 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 		pdpClient,
 	)
 
-	// SP validation
-	err = spDynamic.ValidateServicePrincipal(ctx, spClientCred)
-	if err != nil {
-		return err
+	if dv.oc.Properties.PlatformWorkloadIdentityProfile == nil || dv.oc.Properties.ServicePrincipalProfile != nil {
```

Comment:
cc @slawande2 - here's another place we want to be able to use a shared function to determine if a cluster is workload identity enabled or not.

Reply:
Should we move over to the shared function here?